src/core/ext/filters/client_channel/client_channel.cc

```diff
@@ -656,7 +657,15 @@ class ClientChannel::SubchannelWrapper : public SubchannelInterface {
       // Ignore update if the parent WatcherWrapper has been replaced
       // since this callback was scheduled.
       if (watcher_ != nullptr) {
-        watcher_->OnConnectivityStateChange(state_change.state);
+        // Propagate status only in state TF.
+        // We specifically want to avoid propagating the status for
```

Comment:
What would happen if propagate this anyway?

Reply:
I don't know that it would necessarily break anything, but it would definitely not be the intended semantic that we want to expose to the LB policy API.  The intended semantic is that a non-OK status is returned only in state TRANSIENT_FAILURE.

The only reason that this is an issue is the special-case hack that we added in #29428 to have the real subchannel return a non-OK status in IDLE state so that it has a way of returning the keepalive info.  We don't want to expose that hack to the subchannel interface that is used by the LB policies.